### PR TITLE
ci: Remove publish unit tests

### PR DESCRIPTION
**Pull request by mathiasvr**
This removes the CI step that publishes unit test results, because it doesn't really work with PRs from forked repos, and worse it fails on PRs from e.g. dependabot.

It looks like it possible to fix this by moving it to a separate workflow, but I'm not sure how useful that is.

I think it's best to remove this now and maybe look into this later, but this is not an important step but more a convenience.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by Asizon — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by McGiverGim — APPROVED** [review]
It was beautiful, but unfortunately we don't use too much the unit tests... So to me is ok to remove it, make it work complicates the workflow according the instructions.
But I'm open to include it later.

**Review by haslinghuis — APPROVED** [review]

**Pull request merged**
